htdocs/core/lib/functions.lib.php

```diff
@@ -4431,7 +4431,7 @@ function print_barre_liste($titre, $page, $file, $options = '', $sortfield = '',
 	if ($picto == 'setup') $picto = 'title_setup.png';
 	if (($conf->browser->name == 'ie') && $picto == 'generic') $picto = 'title.gif';
 	if ($limit < 0) $limit = $conf->liste_limit;
-	if ($savlimit != 0 && (($num > $limit) || ($num == -1) || ($limit == 0)))
+	if ($savlimit != 0 && (($num > $limit) || ($num != -1) || ($limit == 0)))
```

Comment:
num = -1 when parm is not provided (so for backward compatibility). This means navigation will be broken for all use of this method. For new usage, it is never -1.